src/Stratis.Bitcoin.Features.Wallet/Wallet.cs

```diff
@@ -380,6 +405,47 @@ public HdAccount AddNewAccount(string password, string encryptedSeed, byte[] cha
 
             return newAccount;
         }
+
+        /// <summary>
+        /// Adds an account to the current account root using extended public key.
+        /// </summary>
+        /// <param name="accountExtPubKey">The extended public key for the account.</param>
+        /// <param name="accountIndex">The zero-based account index.</param>
+        /// <param name="network">The network for which this account will be created.</param>
+        /// <param name="accountCreationTime">Creation time of the account to be created.</param>
+        /// <returns>A new HD account.</returns>
+        public HdAccount AddNewAccount(ExtPubKey accountExtPubKey, int accountIndex, Network network, DateTimeOffset accountCreationTime)
+        {
+            ICollection<HdAccount> accounts = this.Accounts;
```

Comment:
I can see a race condition here. 

So if we look at how it's handled on 379- there is a safeguard against a race condition when this method is executed twice in parallel. 

In your case you'll be getting an exception: collection was modified during the enumaration from time to time. 

If you have time: please introduce a lock to the wallet. If not- do a quickfix like on 379

Reply:
oh cool, good spot - let me look at it